mirgecom/euler.py

```diff
@@ -171,8 +171,12 @@ def _get_wavespeed(dim, eos, cv: ConservedVars):
     return actx.np.sqrt(np.dot(v, v)) + eos.sound_speed(cv)
 
 
-def _facial_flux(discr, eos, q_tpair):
-    """Return the flux across a face given the solution on both sides *q_tpair*."""
+def _facial_flux(discr, eos, q_tpair, local=False):
+    """Return the flux across a face given the solution on both sides *q_tpair*.
+
+    If *local* is *True*, then the returned flux is NOT projected to "all_faces,"
+    but instead remains on *q_tpair.dd*.
```

Comment:
`result_dd="all_faces"` instead of `local`? This would avoid cluttering the code with a conditional.

Reply:
Hrm.   It doesn't seem to like it when I try to project to the same dd:
```
( ... )
>               bnd_flux = split_conserved(dim, _facial_flux(discr, eos,
                                                             bnd_pair, dd=bnd_pair.dd))

test_bc.py:172: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../mirgecom/euler.py:201: in _facial_flux
    return discr.project(q_tpair.dd, dd, flux_weak)
../../grudge/grudge/eager.py:89: in project
    return obj_array_vectorize(
/usr/workspace/wsa/xpacc/Users/mtcampbe/CEESD/Quartz/Install/conda/envs/ceesd.master/lib/python3.8/site-packages/pytools/obj_array.py:141: in obj_array_vectorize
    result[i] = f(ary[i])
../../grudge/grudge/eager.py:90: in <lambda>
    lambda el: self.project(src, tgt, el), vec)
../../grudge/grudge/eager.py:92: in project
    return self.connection_from_dds(src, tgt)(vec)
/usr/workspace/wsa/xpacc/Users/mtcampbe/CEESD/Quartz/Install/conda/envs/ceesd.master/lib/python3.8/site-packages/pytools/__init__.py:684: in wrapper
    result = function(obj, *args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <grudge.eager.EagerDGDiscretization object at 0x2aaaac6b30a0>
from_dd = DOFDesc(<DTAG_BOUNDARY(<class 'meshmode.mesh.BTAG_ALL'>)>, QTAG_NONE)
to_dd = DOFDesc(<DTAG_BOUNDARY(<class 'meshmode.mesh.BTAG_ALL'>)>, QTAG_NONE)
( ... )
        else:
>           raise ValueError("cannot interpolate from: " + str(from_dd))
E           ValueError: cannot interpolate from: DOFDesc(<DTAG_BOUNDARY(<class 'meshmode.mesh.BTAG_ALL'>)>, QTAG_NONE)

../../grudge/grudge/discretization.py:275: ValueError
============================================ short test summary info =============================================

```